auth-web/src/components/auth/staff/admin/BNRequestManager.vue

```diff
@@ -0,0 +1,143 @@
+            <td>{{ item.requestType }}</td>
+            <td>
+              <v-icon v-if="item.isProcessed" color="green" class="names-text pr-1" small>mdi-check</v-icon>
+              <v-icon v-else color="red" class="names-text pr-1" small>mdi-close</v-icon>
+            </td>
+            <td>
+              <v-icon v-if="item.isAdmin" color="green" class="names-text pr-1" small>mdi-check</v-icon>
+              <v-icon v-else color="red" class="names-text pr-1" small>mdi-close</v-icon>
+            </td>
+            <td class="action-cell">
+              <div class="actions" :id="`action-menu-${index}`">
+                <span class="open-action">
+                  <v-btn
+                    small
+                    color="primary"
+                    min-width="5rem"
+                    min-height="2rem"
+                    class="open-action-btn"
+                    @click="showResubmitRequestDialog(item)"
+                  >
+                    Edit
+                  </v-btn>
+                </span>
+              </div>
+            </td>
+          </tr>
+        </template>
+      </v-data-table>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Watch } from 'vue-property-decorator'
+import { RequestTracker, ResubmitBNRequest } from '@/models/request-tracker'
+import ResubmitRequestDialog from '@/components/auth/staff/admin/ResubmitRequestDialog.vue'
+import Vue from 'vue'
+import { namespace } from 'vuex-class'
+
+const BusinessModule = namespace('business')
+
+@Component({
+  components: {
+    ResubmitRequestDialog
+  }
+})
+export default class BNRequestManager extends Vue {
+  @BusinessModule.Action('getBNRequests')
+  private readonly getBNRequests!: (businessIdentifier: string) => Promise<RequestTracker[]>
+
+  @BusinessModule.Action('getRequestTracker')
+  private readonly getRequestTracker!: (requestTrackerId: number) => Promise<RequestTracker>
+
+  @BusinessModule.Action('resubmitBNRequest')
+  private readonly resubmitBNRequest!: (resubmitBNRequest: ResubmitBNRequest) => Promise<boolean>
+
+  @Prop({ default: undefined }) businessIdentifier: string
+
+  protected headers = [
+    { text: 'Request Type', align: 'start', value: 'requestType', sortable: false, show: true },
+    { text: 'Processed', value: 'isProcessed', sortable: false, show: true },
+    { text: 'Admin Request', value: 'isAdmin', sortable: false, show: true },
+    { text: 'Actions', align: 'end', value: 'action', sortable: false, show: true }
+  ]
+
+  protected bnRequests: RequestTracker[] = []
+  protected requestDetails: RequestTracker = null
+  protected resubmitRequestDialog = false
+
+  @Watch('businessIdentifier', { deep: true, immediate: true })
+  async businessIdentifierChange () {
+    this.bnRequests = await this.getBNRequests(this.businessIdentifier)
+  }
+
+  private async showResubmitRequestDialog (item: RequestTracker): Promise<void> {
+    this.requestDetails = await this.getRequestTracker(item.id)
```

Comment:
As in the other component, `getRequestTracker()` should either return null if it encounters an error, or you need to handle it here.

I noticed that line 5 de-references "requestDetails" so it should be updated to handle null.

Reply:
I have already included `v-if="requestDetails"` in line 3

Reply:
OK, this seems to work because you initialize requestDetails to null and you also reset it when hiding the dialog.

It's good practise to handle exceptions from network calls anyway. Since you don't, here, line 98 will cause the JS thread to terminate, so line 99 won't get run -- the dialog won't display if there was a network error. This is a bug.